coderd/database/dbfake/dbfake.go

```diff
@@ -744,6 +751,37 @@ func (q *FakeQuerier) AcquireProvisionerJob(_ context.Context, arg database.Acqu
 	return database.ProvisionerJob{}, sql.ErrNoRows
 }
 
+func (q *FakeQuerier) ActivityBumpWorkspace(ctx context.Context, workspaceID uuid.UUID) error {
+	err := validateDatabaseType(workspaceID)
+	if err != nil {
+		return err
+	}
+
+	q.mutex.Lock()
+	defer q.mutex.Unlock()
+
+	workspace, err := q.getWorkspaceByIDNoLock(ctx, workspaceID)
+	if err != nil {
+		return err
+	}
+	latestBuild, err := q.getLatestWorkspaceBuildByWorkspaceIDNoLock(ctx, workspaceID)
+	if err != nil {
+		return err
+	}
+
+	for i := range q.workspaceBuilds {
+		if q.workspaceBuilds[i].BuildNumber != latestBuild.BuildNumber {
+			continue
+		}
+		newDeadline := q.workspaceBuilds[i].Deadline.Add(time.Duration(workspace.Ttl.Int64))
+		q.workspaceBuilds[i].UpdatedAt = latestBuild.UpdatedAt
+		q.workspaceBuilds[i].Deadline = minTime(newDeadline, q.workspaceBuilds[i].MaxDeadline)
+		return nil
+	}
+
+	return sql.ErrNoRows
+}
```

Comment:
We do not need more logic in here? Like the job state and everything to match the sql?

Reply:
That's old code, see https://github.com/coder/coder/pull/9652/files#diff-17a8f755cac54c55dd9681d52cfeea1f80fe7b75191bda5c9229cc31496c6fc5R777-R792